### ISSUE TITLE
Allow variables and types with same name

### ISSUE THREAD

**Issue by spencerhakim** [eslint/typescript-eslint-parser#359]
**What version of TypeScript are you using?**
2.4.2

**What version of `typescript-eslint-parser` are you using?**
5.0.1

**What code were you trying to parse?**

```
class MyFoo {}
export const Foo = MyFoo;
export type Foo = MyFoo;
```

**What did you expect to happen?**
No warnings/errors, [this is valid TS](https://github.com/Microsoft/TypeScript/issues/10058#issuecomment-236458961).

**What happened?**
The `no-redeclare` rule was triggered since `Foo` had already been defined once.

**Comment by JamesHenry** [comment]
Thanks a lot for your patience on this @spencerhakim!

This is down to the way in which we currently handle `type` declarations.

**Comment by j-f1** [comment]
As a workaround, you can do

```ts
class MyFoo {}

export { MyFoo as Foo }
```

**Comment by JamesHenry** [comment]
Closing in favour of a single issue here: https://github.com/eslint/typescript-eslint-parser/issues/414
